src/alcaramba/Drawable/Game/Game.gd

```diff
@@ -45,25 +52,49 @@ func _on_tile_card_selected(card_node: TileCardDrawable):
 func _on_spare_selected(card_node: TileCardDrawable):
 	# send selected tile to town for placement overlay
 	_town.receive_tile(card_node)
+	_cancel_place_tile_button.disabled = false
+	
+	if _cancel_place_tile_button.is_connected("pressed", _spare_tiles, "_on_place_tile_canceled"):
+		_cancel_place_tile_button.disconnect("pressed", _spare_tiles, "_on_place_tile_canceled")
+	_cancel_place_tile_button.connect("pressed", _spare_tiles, "_on_place_tile_canceled", [card_node], CONNECT_ONESHOT)
+
 
 # if tile is placed add it to town stack and remove selection
 func _on_tile_placed(tile: TileCard):
+	_cancel_place_tile_button.disabled = true
+	
 	# Disconnect button so tile cannot be added to spares after placing them
 	if _spare_tile_add_button.is_connected("pressed", _spare_tiles, "_on_add_to_spares"):
 		_spare_tile_add_button.disconnect("pressed", _spare_tiles, "_on_add_to_spares")
 	
 	if _selected_tile != null:
 		_selected_tile.clear()
```

Comment:
_selected_tile is not null when placing from spare. Works anyway but removal from spare does not happen in else part

Reply:
As discussed, let's rework this later